feat: added ability to search / filter out components

cadamsdev opened the pull request:
Media

![msedge_V4USxjE96J](https://github.com/user-attachments/assets/0e4881d1-ea6f-47b0-85fd-b6b65c265017)

kof reviewed (approved):
amazing!

Approving to allow deployment, still in review process, will add more items

- [x] now we need to also add fuzzy search, check out how we use "matchSorter"
- [ ] need to fix SearchField ESC handling, seems broken, in fonts manager it should clear the seach first, not close the popover right away
- [x] add a separator line after search input
- [x] add autoFocus, so that when opening, input has caret right away

cadamsdev commented on apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx:
```diff
@@ -74,16 +89,56 @@ export const TabContent = ({ publish, onSetActiveTab }: TabContentProps) => {
-                  {(metaByCategory.get(category) ?? [])
+                  {(metaByCategory.get(categoryGroup.category) ?? [])
                     .filter((meta: WsComponentMeta) => {
+                      const component = componentNamesByMeta.get(meta);
+                      if (
+                        searchText !== "" &&
+                        !component
+                          ?.toLowerCase()
+                          .includes(searchText.toLowerCase())
+                      ) {
+                        return false;
+                      }
```
TODO delete this part not needed

cadamsdev commented:
> amazing!
> 
> Approving to allow deployment, still in review process, will add more items
> 
> * [ ]  now we need to also add fuzzy search, check out how we use "matchSorter"
> * [ ]  need to fix SearchField ESC handling, seems broken, in fonts manager it should clear the seach first, not close the popover right away
> * [ ]  add a separator line after search input
> * [ ]  add autoFocus, so that when opening, input has caret right away

- Added fuzzy search
- Added separator line after search input
- Added autoFocus

kof commented:
Merged, follow up PR will:
- fix search field
- allow using keyboard arrows to navigate
- remove categories when searching
